data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/JsonViewQueryResultMapperFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.runtime.mapper.sql;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.model.runtime.RuntimePersistentEntity;
+import io.micronaut.data.runtime.mapper.ResultReader;
+import io.micronaut.serde.ObjectMapper;
+
+import java.util.function.BiFunction;
+
+/**
+ * The factory for creating JSON view {@link JsonQueryResultMapper} for given SQL dialect.
+ *
+ * @author radovanradic
+ * @since 4.0.0
+ *
+ * @param <T> the entity type
+ * @param <RS> the result set type
+ * @param <R> the result type
+ */
+public interface JsonViewQueryResultMapperFactory<T, RS, R> {
```

Comment:
I would use `view` in the name, as it doesn't matter if you use the view.
It might be better to have something like `QueryResultMapperRegistry` and maybe `QueryResultMapperProvider` with some method to check if it can be used. In the future we can support different types like XML etc.

Reply:
I see. Main issue for me was - do we configure JSON deserialization on column level or entity/result level. I mean users can select JSON in oracle and return Oracle JSON object but was not sure how we can mark that for individual field.
This is why I got stuck with JsonView as we need to  implement that first.

Reply:
I have added some level of abstraction to create query result mapper in `AbstractSqlRepositoryOperations`